### PR TITLE
Add details around configuration and options flow

### PR DESCRIPTION
**Pull request by kit-klein**
## Proposed change

This PR adds missing details for the Konnected.io configuration flow based setup that was released in 0.106.  

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
- This PR fixes or closes issue: #12203

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/konnected.markdown [hunk 1]
```diff
@@ -18,13 +18,19 @@ The `konnected` integration lets you connect wired sensors and switches to a Kon
...
+</div>
+
 The integration currently supports the following device types in Home Assistant:
 
 - Binary Sensor: Wired door and window sensors, motion detectors, glass-break detectors, leak sensors, smoke & CO detectors or any open/close switch.
 - Switch: Actuate a siren, strobe, buzzer or relay module.
 - Sensor: Periodic measurements from DHT temperature/humidity sensors and DS18B20 temperature sensors.
 
-This integration uses the [SSDP](/integrations/ssdp) integration, which must be enabled for device discovery to work. If you don't want to use discovery, set the _host_ and _port_ for each device in the description.
+This integration uses the [SSDP](/integrations/ssdp) integration, which must be enabled for device discovery to work. If you don't want to use ssdp you'll need to manually provide the IP and Port information for each Konnected Panel. The IP/Port info can be found using the Konnected mobile app.
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
This integration uses the [SSDP](/integrations/ssdp) integration, which must be enabled for device discovery to work. If you don't want to use SSDP you'll need to manually provide the IP and Port information for each Konnected Panel. The IP/Port info can be found using the Konnected mobile app.
```

File: source/_integrations/konnected.markdown [hunk 2]
```diff
@@ -34,14 +40,53 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
...
 
-If you prefer you can also utilize a `konnected` section in the `configuration.yaml` file that specifies the Konnected devices on the network and the sensors or actuators attached to them.  If using `configuration.yaml` the configuration will automatically be imported and used to start a configuration flow. **Note that you must still complete the configuration/options flow before the configuration entry will be finalized.**
+Selecting one of these discovered panels will launch a configuration flow to guide you through a setup process to initially connect the panel and generate a configuration entry. If your panel was discovered via SSDP you shouldn't need any information to complete the configuration flow - simply confirm that the information is correct. If the configuration flow prompts you for IP/Port you'll need to enter it. IP/Port info can be found using the Konnected mobile app.
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
Selecting one of these discovered panels will launch a configuration flow to guide you through a setup process to initially connect the panel and generate a configuration entry. If your panel was discovered via SSDP, you shouldn't need any information to complete the configuration flow - simply confirm that the information is correct. If the configuration flow prompts you for IP/Port, you'll need to enter it. IP/Port info can be found using the Konnected mobile app.
```

File: source/_integrations/konnected.markdown [hunk 3]
```diff
@@ -34,14 +40,53 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
...
 
-Details of the configuration fields and values can be found below - these apply to both the configuration flow and the YAML.
+<div class='note info'>
+
+If you have an existing `configuration.yaml` completing the configuration flow will do an one time import of the settings contained in `configuration.yaml`. Once the import creates a configuration entry the Konnected section of the `configuration.yaml` is no longer used - it is recommended to remove the konnected section of `configuration.yaml` and after the import occurs. Any future changes to settings should occur via the Options Flow.
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
If you have an existing `configuration.yaml` completing the configuration flow will do a one time import of the settings contained in `configuration.yaml`. Once the import creates a configuration entry the Konnected section of the `configuration.yaml` is no longer used - it is recommended to remove the `konnected` section of `configuration.yaml` and after the import occurs. Any future changes to settings should occur via the options provided in the Home Assistant web interface.
```

File: source/_integrations/konnected.markdown [hunk 4]
```diff
@@ -34,14 +40,53 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
...
+
+If you want to retain `configuration.yaml` and need to re-import any changes or updates you will need to delete the entry in **Configuration** -> **Integrations** -> **Configured** and repeat the configuration flow.
+
+</div>  
+
+Once the configuration flow is completed you'll see a Konnected.io entry in **Configuration** -> **Integrations** => **Configured**.  If you imported settings from `configuration.yaml` you are now done! If you are setting up a new Konnected Alarm Panel or modifying settings you'll need to utilize the Options Flow to configure zone behavior.
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
Once the configuration flow is completed you'll see a Konnected.io entry in **Configuration** -> **Integrations** => **Configured**.  If you imported settings from `configuration.yaml` you are now done! If you are setting up a new Konnected Alarm Panel or modifying settings, you'll need to utilize the Options Flow to configure zone behavior.
```

File: source/_integrations/konnected.markdown [hunk 5]
```diff
@@ -34,14 +40,53 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
...
+#### Binary Sensor:
+
+**Binary Sensor Type:** The type of sensor connected to the zone.
+**Name (optional)** The friendly name for the entity associated with the zone.
+**Invert the open/close state:** Inverts the open/closed meaning of a binary sensor circuit. Commonly needed for normally open wired smoke alarm circuits.
+
+#### Digital Sensor:
+
+**Sensor Type:** The type of sensor connected to the zone - either `dht` or `ds18b20`.
+**Name (optional)** The friendly name for the entities associated with the zone
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
**Name (optional)** The friendly name for the entities associated with the zone.
```

File: source/_integrations/konnected.markdown [hunk 6]
```diff
@@ -34,14 +40,53 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
...
+
+</div>  
+
+Once the configuration flow is completed you'll see a Konnected.io entry in **Configuration** -> **Integrations** => **Configured**.  If you imported settings from `configuration.yaml` you are now done! If you are setting up a new Konnected Alarm Panel or modifying settings you'll need to utilize the Options Flow to configure zone behavior.
+
+### Options Flow
+
+The Options Flow for each panel can be accessed by selecting the entry in **Configuration** -> **Integrations** => **Configured** and then clicking on the gear icon in the upper right corner. You can reconfigure settings in the Options Flow at any time.  Once you complete the Options Flow the settings will be immediately applied.
 
+Options Flow starts by having you configure the general behavior of each zone. You need to specify `Disabled`, `Binary Sensor`, `Digital Sensor`, or `Switchable Output` for each zone.  After that you'll be prompted, for each zone that is not disabled, to configure details of the zones behavior. All zones will allow entry of a Name. Additional fields depend on how you configured the general behavior of the zone.
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
Options Flow starts by having you configure the general behavior of each zone. You need to specify `Disabled`, `Binary Sensor`, `Digital Sensor`, or `Switchable Output` for each zone.  After that, you'll be prompted, for each zone that is not disabled, to configure details of the zones' behavior. All zones will allow entry of a Name. Additional fields depend on how you configured the general behavior of the zone.  
```

File: source/_integrations/konnected.markdown [hunk 7]
```diff
@@ -34,14 +40,53 @@ Konnected devices communicate with Home Assistant over your local LAN -- there i
...
+**Name (optional)** The friendly name for the entities associated with the zone
+**Poll Interval (optional):** How often in minutes to poll the sensor for updates.
+
+#### Switchable Output:
+
+**Name: (optional)** The friendly name for the entity associated with the zone
+**Output when on:** The state of the switch when activated.
+**Pulse Duration (optional):** The duration in ms to pulse the switch once activated
+**Pause between pulses (optional):** The duration in ms to wait between pulses when activated
+**Times to repeat (optional):** The number of times to repeat the pulse each time the switch is activated
```
Comment (frenck) [inline source/_integrations/konnected.markdown]:
```suggestion
**Name: (optional)** The friendly name for the entity associated with the zone.
**Output when on:** The state of the switch when activated.
**Pulse Duration (optional):** The duration in ms to pulse the switch once activated.
**Pause between pulses (optional):** The duration in ms to wait between pulses when activated.
**Times to repeat (optional):** The number of times to repeat the pulse each time the switch is activated.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Some small things provided as suggestions.

In general, I think we should avoid the terms "configuration flow" or "options flow", these are terms used internally as developers. However, users aren't familiar with those terms, nor are they listed anywhere. In general, we refer to the "UI", "web interface" or the Home Assistant (web) frontend.

**Comment by kit-klein** [comment]
> Some small things provided as suggestions.
> 
> In general, I think we should avoid the terms "configuration flow" or "options flow", these are terms used internally as developers. However, users aren't familiar with those terms, nor are they listed anywhere. In general, we refer to the "UI", "web interface" or the Home Assistant (web) frontend.

Thank you for the clean up suggestions.  I'll go through and replace up some of the technical terms (configuration.yaml) as well.

**Comment by kit-klein** [comment]
> Some small things provided as suggestions.
> 
> In general, I think we should avoid the terms "configuration flow" or "options flow", these are terms used internally as developers. However, users aren't familiar with those terms, nor are they listed anywhere. In general, we refer to the "UI", "web interface" or the Home Assistant (web) frontend.

@frenck The latest commit removes the technical terminology.  Let me know if you see anything else.

**Review by frenck — APPROVED** [review]
Nice! @kit-klein 👍

**Pull request merged**